src/js_types.hpp

```diff
@@ -507,7 +510,7 @@ inline bool Value<T>::is_valid_for_property_type(ContextType context, const Valu
             && (type != PropertyType::Object || list->get_object_schema().name == object_type);
     };
 
-    if (!realm::is_array(type)) {
+    if (!realm::is_array(type) && !realm::is_set(type)) {
```

Comment:
It seems that there is no explicit test here for if `set` `is_valid_for_property_type`.  But perhaps thats legitimate, I do not have much knowledge on how this function is being used.

Reply:
`is_valid_for_property_type` is invoked when elements are inserted into the Set.  It asserts that the new element conforms to the type of Set it is inserted into.